File: src/main/java/schemabuilder/processor/pipelines/parsing/datafetchers/DataFetcherCostMap.java
package schemabuilder.processor.pipelines.parsing.datafetchers;

import graphql.schema.DataFetcher;
import java.util.HashMap;
import java.util.Map;

public class DataFetcherCostMap {

    private static final Map<DataFetcher, Integer> map = new HashMap<>();

    public static int getCostFor(DataFetcher dataFetcher) {
        System.out.println("Getting cost for: " + dataFetcher);
        return map.getOrDefault(dataFetcher, 0);
    }

    public static void setCostFor(DataFetcher dataFetcher, int cost) {
        System.out.println("Setting cost for: " + dataFetcher);
        map.put(dataFetcher, cost);
    }
}

File: src/main/java/schemabuilder/processor/pipelines/parsing/datafetchers/DataFetcherCostMap.kt
package schemabuilder.processor.pipelines.parsing.datafetchers

import graphql.schema.DataFetcher
import java.util.*

object DataFetcherCostMap {

    private val map: MutableMap<DataFetcher<*>, Int> = HashMap()

    @kotlin.jvm.JvmStatic
    fun getCostFor(dataFetcher: DataFetcher<*>): Int {
        return map.getOrDefault(dataFetcher, 0)
    }

    @kotlin.jvm.JvmStatic
    fun setCostFor(dataFetcher: DataFetcher<*>, cost: Int) {
        map[dataFetcher] = cost
    }
}